test/screenshot/screenshot.js

```diff
@@ -1,62 +1,77 @@
+    this.imagePath_ = `${urlPath}.golden.png`;
+    this.testImagePath_ = `${urlPath}.test.png`;
+    this.diffPath_ = `${urlPath}.diff.png`;
     // TODO allow clients to specify capture-chrome options, like viewport size
   }
 
   capture() {
-    test(this.urlPath_, () => {
-      const url = 'http://localhost:8080/' + this.urlPath_;
-      return this.checkStatusCode_(url,
-        capture({
-          url,
-        }).then((screenshot) => {
-          fs.writeFileSync(
-            './test/screenshot/' + this.imagePath_,
-            screenshot);
-        })
-      );
+    test(this.urlPath_, async () => {
+      const url = `http://localhost:8080/${this.urlPath_}`;
+      const imagePath = `./test/screenshot/${this.imagePath_}`;
+      await this.createScreenshotTask_(url, imagePath);
+      return;
     });
   }
 
   diff() {
-    test(this.urlPath_, () => {
-      const capturePromise = capture({
-        url: 'http://localhost:8080/' + this.urlPath_,
-      });
-      const readPromise = readFilePromise(
-          './test/screenshot/' + this.imagePath_);
-      return Promise.all([capturePromise, readPromise])
-      .then(function([newScreenshot, oldScreenshot]) {
-        return new Promise(function(resolve) {
-            const onComplete = function(data) {
-              assert.isBelow(Number(data.misMatchPercentage), 0.01);
-              resolve();
-            };
-            resemble(newScreenshot)
-            .compareTo(oldScreenshot)
-            .onComplete(onComplete);
-        });
-      });
+    test(this.urlPath_, async () => {
+      const url = `http://localhost:8080/${this.urlPath_}`;
+      const imagePath = `./test/screenshot/${this.imagePath_}`;
+      const testImagePath = `./test/screenshot/${this.testImagePath_}`;
+      const diffPath = `./test/screenshot/${this.diffPath_}`;
+
+      const [newScreenshot, oldScreenshot] = await Promise.all([
+        this.createScreenshotTask_(url, testImagePath),
+        readFilePromise(imagePath),
+      ]);
+
+      const options = {
+        output: {
+          errorColor: {
+            red: 255,
+            green: 0,
+            blue: 255,
+            alpha: 150,
+          },
+          errorType: 'movementDifferenceIntensity',
+          transparency: 0.3,
+        },
+        ignore: ['antialiasing'],
+      };
+
+      const data = await compareImages(newScreenshot, oldScreenshot, options);
+
+      await writeFilePromise(diffPath, data.getBuffer());
+
+      assert.isAtMost(Number(data.misMatchPercentage), 0.01);
+      return;
     });
   }
 
-  checkStatusCode_(url, success) {
-    return new Promise((resolve) => {
-      get(url, (res) => {
-        const {statusCode} = res;
-        if (statusCode === 200) {
-          resolve(success);
-        }
-      });
-    });
+  async createScreenshotTask_(url, path) {
+    let image;
+    const browser = await puppeteer.launch();
+    const page = await browser.newPage();
+    await page.goto(url);
+    if (path) {
```

Comment:
You seem to always pass `path` into `createScreenshotTask` so you don't need this if check